csrc/python_frontend/fusion_record.h

```diff
@@ -381,6 +382,96 @@ struct ReshapeOpRecord : RecordFunctor {
   std::vector<int64_t> new_shape_;
 };
 
+struct PadOpRecord : RecordFunctor {
+  PadOpRecord(
+      std::vector<State> _args,
+      std::vector<State> _outputs,
+      std::vector<int64_t>& pad_widths)
+      : RecordFunctor(
+            std::move(_args),
+            std::move(_outputs),
+            "ops.pad",
+            RecordType::PadOp),
+        pad_widths_(std::move(pad_widths)) {}
+  virtual ~PadOpRecord() = default;
+  virtual RecordFunctor* clone() final {
+    return new PadOpRecord(*this);
+  }
+
+  virtual size_t hash() const final {
```

Comment:
nitpick, as with other ops, can we also add a comment on the hash bit allocation.

Reply:
Got me to look at my approach here again. I'm now circshifting each pad width by 4 bits now before XORing. Prevents e.g. [1, 1, 1, 0] from conflicting with [0, 1], [1,0], [0, 1, 1, 1] etc.